packages/components/data-table/src/data-table.js

```diff
@@ -1,57 +1,102 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { filterDataAttributes } from '@commercetools-uikit/utils';
+import {
+  getColumnWidthsArray,
+  getGridTemplateColumnsStyle,
+} from './column-size-utils';
 import { TableGrid, Header, Body, Row, Footer } from './data-table.styles';
 import { HeaderCell, FooterCell } from './cell';
 import DataRow from './data-row';
 
-const DataTable = (props) => (
-  <TableGrid
-    {...filterDataAttributes(props)}
-    columns={props.columns}
-    maxWidth={props.maxWidth}
-    maxHeight={props.maxHeight}
-  >
-    <Header>
-      <Row>
-        {props.columns.map((column) => (
-          <HeaderCell
-            key={column.key}
-            isCondensed={props.isCondensed}
-            /* Sorting Props */
-            onClick={props.onSortChange}
-            sortedBy={props.sortedBy}
-            columnKey={column.key}
-            isSortable={column.isSortable}
-            shouldWrap={props.wrapHeaderLabels}
-            sortDirection={props.sortDirection}
-            disableHeaderStickiness={props.disableHeaderStickiness}
-          >
-            {column.label}
-          </HeaderCell>
-        ))}
-      </Row>
-    </Header>
-    <Body>
-      {props.rows.map((row, rowIndex) => (
-        <DataRow {...props} row={row} key={rowIndex} rowIndex={rowIndex} />
-      ))}
-    </Body>
-    {props.footer && (
-      <Footer>
+const DataTable = (props) => {
+  const [columnSizing, setcolumnSizing] = React.useState({
+    isManualControlled: false,
+    sizes: getColumnWidthsArray(props.columns),
+  });
+  const tableRef = React.useRef();
+
+  const gridTemplateColumnsStyle = getGridTemplateColumnsStyle(
+    columnSizing.sizes
+  );
+
+  const onColumnResizeStart = () => {
+    // when a user starts resizing a column, all columns sizes stop being responsive and become manually controlled
+    if (!columnSizing.isManualControlled) {
+      const renderedLayoutSizes = [];
+      tableRef.current.querySelectorAll('th').forEach((header, index) => {
+        renderedLayoutSizes[index] = header.clientWidth;
+      });
+      setcolumnSizing({
+        isManualControlled: true,
+        sizes: renderedLayoutSizes,
+      });
+    }
+  };
```

Comment:
My main wonder here is if we should hoist state management into a `useReducer` and put it onto the context if needed. It's a separate, additional suggestion to the other suggestion of using a reducer.